lib/include/tick/array/array.h

```diff
@@ -45,6 +45,9 @@ class SArray;
  */
 template <typename T>
 class Array : public BaseArray<T> {
+  template <class T1>
```

Comment:
Why T1 here and T on line 415? 
Can't this be mixed with `tick::TemporaryLog<E>& operator<<(tick::TemporaryLog<E>& log, const Array<T>& arr)`?

Reply:
T1 is required as it's a thing called "template shadowing" 

it's good in this case to let the compiler know that May be a different type(not that will be)

Reply:
Could it be? Because if it will never be I rather think it makes the code more difficult to understand